src/components/accessories/patientTriage/patientTriageForm/PatientTriageForm.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { FunctionComponent, useCallback, useState } from 'react';
+import { useFormik } from "formik";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../../libraries/formDataHandling/functions";
+import DateField from '../../dateField/DateField';
+import { object } from 'yup';
+import { TProps } from './types';
+import ConfirmationDialog from '../../confirmationDialog/ConfirmationDialog';
+import TextButton from '../../textButton/TextButton';
+import SmallButton from '../../smallButton/SmallButton';
+import warningIcon from "../../../../assets/warning-icon.png";
+import TextField from '../../textField/TextField';
+import has from 'lodash.has';
+import get from 'lodash.get';
+import SelectField from '../../selectField/SelectField';
+import "./styles.scss"
+
+const PatientTriageForm: FunctionComponent<TProps> = ({
+  fields,
+  onSubmit,
+  submitButtonLabel,
+  resetButtonLabel,
+  isLoading,
+}) => {
+
+  const validationSchema = object({
+    // TODO  
+  });
+
+  const initialValues = getFromFields(fields, "value");
+  const options = getFromFields(fields, "options");
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      const formattedValues = formatAllFieldValues(fields, values);
+      onSubmit(formattedValues);
+    },
+  });
+
+  const { setFieldValue, handleBlur } = formik;
+
+  const dateFieldHandleOnChange = useCallback(
+    (value) => {
+      setFieldValue("triageDate", value);
```

Comment:
Maybe we could parameterize the `setFieldValue`'s first argument (which in this case is a hardcoded "triageDate") with a curried function, so we can get this callback generalized.

Reply:
I open a separate branch in order to fix this.